src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/RestoreCommand.cs

```diff
@@ -123,6 +124,11 @@ internal static class RestoreCommand
                             restoreContext.FallbackRuntimes.UnionWith(defaultRuntimes);
                         }
 
+                        if (restoreContext.DisableParallel)
+                        {
+                            HttpSourceResourceProvider.Throttle = new SemaphoreSlimThrottle(new SemaphoreSlim(1));
```

Comment:
This statement is repeated multiple times. Can we actually hide the details while having convenient constructor with no parameters?